akka-docs/src/main/paradox/typed/persistence-durable-state.md

```diff
@@ -0,0 +1,391 @@
+
+The database specific implementations can be added to existing Akka Persistence plugin implementations, starting with the JDBC plugin. The plugin would serialize the state and store as a blob with the persistenceId as the primary key. 
+
+## Example and core API
+
+Let's start with a simple example that models a counter using an Akka persistent actor. The minimum required for a @apidoc[DurableStateBehavior] is:
+
+Scala
+:  @@snip [DurableStatePersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala) { #structure }
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #structure }
+
+The first important thing to notice is the `Behavior` of a persistent actor is typed to the type of the `Command`
+because this is the type of message a persistent actor should receive. In Akka this is now enforced by the type system.
+
+The components that make up a `DurableStateBehavior` are:
+
+* `persistenceId` is the stable unique identifier for the persistent actor.
+* `emptyState` defines the `State` when the entity is first created e.g. a Counter would start with 0 as state.
+* `commandHandler` defines how to handle commands and map to appropriate effects e.g. persisting state and replying to actors.
+
+Next we'll discuss each of these in detail.
+
+### PersistenceId
+
+The @apidoc[akka.persistence.typed.PersistenceId] is the stable unique identifier for the persistent actor in the backend
+durabe state store.
+
+@ref:[Cluster Sharding](cluster-sharding.md) is typically used together with `DurableStateBehavior` to ensure
+that there is only one active entity for each `PersistenceId` (`entityId`).
+
+The `entityId` in Cluster Sharding is the business domain identifier of the entity. The `entityId` might not
+be unique enough to be used as the `PersistenceId` by itself. For example two different types of
+entities may have the same `entityId`. To create a unique `PersistenceId` the `entityId` should be prefixed
+with a stable name of the entity type, which typically is the same as the `EntityTypeKey.name` that
+is used in Cluster Sharding. There are @scala[`PersistenceId.apply`]@java[`PersistenceId.of`] factory methods
+to help with constructing such `PersistenceId` from an `entityTypeHint` and `entityId`.
+
+The default separator when concatenating the `entityTypeHint` and `entityId` is `|`, but a custom separator
+is supported.
+
+The @ref:[Persistence example in the Cluster Sharding documentation](cluster-sharding.md#persistence-example)
+illustrates how to construct the `PersistenceId` from the `entityTypeKey` and `entityId` provided by the
+`EntityContext`.
+
+A custom identifier can be created with `PersistenceId.ofUniqueId`.  
```

Comment:
Why do I need to know this here?

Reply:
This is from the event sourcing document .. remove ?